site/bps/BP-25-MovingChecksumToProto.md

```diff
@@ -0,0 +1,27 @@
+---
+title: "BP-25: MovingChecksumToProto--Refactor the checksum part of bookkeeper"
+issue: https://github.com/apache/bookkeeper/issues/1007
+state: "Under Discussion"
+release: "N/A"
+---
+
+### Motivation
+
+Current the checksum implementation is in client module while the checksum semantic is more close to protocol. Moreover, moving the checksum implementation to protocol will avoid server module's dependency to client module when doing checksum in server side.
+
+### Public Interfaces
+
+An internal refactor not affecting public interfaces.
+
+### Proposed Changes
+
+Move the DigestManager and related classes to proto module
```

Comment:
which package are you thinking of "org.apache.bookkeeper.proto" or "org.apache.bookkeeper.proto.checksum"?

Reply:
I think "org.apache.bookkeeper.proto.checksum" is more clear, and the proto module already has lots of classes.  What's your idea?

Reply:
Good Idea; +1. ;

Reply:
@sijie  What was the performance penalty you paid for the check on the bookies? I believe we needed that to avoid corruption at the source (client nic)

Reply:
@ArvinDevel : +1 on `org.apache.bookkeeper.prot.checksum`

@jvrao I think it has slight increased cpu usage but I don't have an exact number of that, because at that time we don't have CRC32C. now we have a native CRC32C implementation, this should be much better now. 

and yes, agreed with you (and I believed we chatted about this long time ago) we have to do the bookie side checksum validation before storing entries.